tests/main/apparmor-prompting-flag-restart/task.yaml

```diff
@@ -0,0 +1,121 @@
+summary: Check that changing the experimental.apparmor-prompting flag causes snapd to restart
+
+details: |
+    This test checks that snapd is restarted when the experimental.apparmor-prompting
+    flag is changed, and that snapd is not restarted when the flag is set to
+    the same value repeatedly. Restarting snapd when the flag changes is
+    necessary so that apparmor kernel and parser features are re-probed, in
+    case support for prompting has changed since snapd last started.
+
+systems:
+  - ubuntu-16.04-*
+  - ubuntu-18.04-*
+  - ubuntu-2*
```

Comment:
should we include core in this too? the feature will be relevant for core desktop

Reply:
I've added core as well, though please confirm which versions of core we should test on (likely only need 24+)